src/common/addlayers/AddLayersDirective.js

```diff
@@ -137,12 +137,14 @@
               layerConfig.CRS = ['EPSG:4326'];
               scope.currentLayer = layerConfig;
               var layer = mapService.createLayerWithFullConfig(layerConfig, scope.currentServerId);
+              layer.extent = layerConfig.extent;
```

Comment:
if you move the code from the MapDirective in here, this line is not needed anymore.